### PR TITLE
feat(sozo/account): legacy bool flag to use old calldata encoding

### PR DESCRIPTION
**Pull request by lambda-0x**
fix: #1418

improvements TODO:
- ~~in `KatanaRunner` `Command` expects the full path of binary~~
- katana's logs should be written to a temporary folder

### CODE REVIEW

**Comment by lambda-0x** [comment]
(git woopsie)

File: crates/sozo/src/commands/options/account.rs [hunk 1]
```diff
@@ -322,4 +324,51 @@ mod tests {
...
+                FieldElement::from_hex_be("0x2").unwrap(),
+                FieldElement::from_hex_be("0x3").unwrap(),
+            ],
+        }];
+
+        // HACK: SingleOwnerAccount doesn't expose a way to check `encoding` type used in struct, so
+        // checking it by encoding a dummy call and checking which method it used to encode the call
+        let account = cmd.account.account(provider, None).await.unwrap();
+        let result = account.encode_calls(&dummy_call);
+        assert!(*result.get(3).unwrap() == FieldElement::from_hex_be("0x2").unwrap());
```
Comment (glihm) [inline crates/sozo/src/commands/options/account.rs:372]:
```suggestion
        // 0x2 is the Calldata len.
        assert!(*result.get(3).unwrap() == FieldElement::from_hex_be("0x2").unwrap());
```

File: crates/sozo/src/commands/options/account.rs [hunk 2]
```diff
@@ -322,4 +324,51 @@ mod tests {
...
+    #[tokio::test]
+    async fn without_legacy_flag_works_as_expected() {
+        let cmd = Command::parse_from(["sozo", "--account-address", "0x0", "--private-key", "0x1"]);
+        let (_runner, provider) = KatanaRunner::new().unwrap();
+        let dummy_call = vec![Call {
+            to: FieldElement::from_hex_be("0x0").unwrap(),
+            selector: FieldElement::from_hex_be("0x1").unwrap(),
+            calldata: vec![
+                FieldElement::from_hex_be("0x2").unwrap(),
+                FieldElement::from_hex_be("0x3").unwrap(),
```
Comment (glihm) [inline crates/sozo/src/commands/options/account.rs:364]:
```suggestion
                FieldElement::from_hex_be("0xf2").unwrap(),
                FieldElement::from_hex_be("0xf3").unwrap(),
```
To clearly identify which field in the encoded data we're testing here.

File: crates/sozo/src/commands/options/account.rs [hunk 3]
```diff
@@ -322,4 +324,51 @@ mod tests {
...
+            ],
+        }];
+
+        // HACK: SingleOwnerAccount doesn't expose a way to check `encoding` type used in struct, so
+        // checking it by encoding a dummy call and checking which method it used to encode the call
+        let account = cmd.account.account(provider, None).await.unwrap();
+        let result = account.encode_calls(&dummy_call);
+        assert!(*result.get(3).unwrap() == FieldElement::from_hex_be("0x0").unwrap());
```
Comment (glihm) [inline crates/sozo/src/commands/options/account.rs:352]:
```suggestion
        // 0x0 is the data offset.
        assert!(*result.get(3).unwrap() == FieldElement::from_hex_be("0x0").unwrap());
```

**Review by glihm — COMMENTED** [review]
Nice hack for testing the actual encoding.

File: crates/sozo/src/commands/options/account.rs [hunk 4]
```diff
@@ -40,6 +40,10 @@ pub struct AccountOptions {
     #[arg(help_heading = "Signer options - KEYSTORE")]
     #[arg(help = "The keystore password. Used with --keystore.")]
     pub keystore_password: Option<String>,
+
+    #[arg(long)]
+    #[arg(help = "Use legacy calldata encoding")]
```
Comment (kariy) [inline crates/sozo/src/commands/options/account.rs:45]: 'Legacy account' may be easier to understand than 'legacy calldata encoding'

**Review by glihm — APPROVED** [review]

**Pull request merged**